training/distributed_training/pytorch/data_parallel/deepspeed/llama2/SMDDP_Deepspeed_example.ipynb

```diff
@@ -0,0 +1,230 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Train Llama2 model using SageMaker Distributed Data Parallel Library (SMDDP) and DeepSpeed\n",
+    "\n",
+    "In this tutorial, we will show how to train or fine-tune the new [LLama2-7B](https://huggingface.co/meta-llama/Llama-2-7b) model.  We will use [DeepSpeed](https://www.deepspeed.ai/training/) ZeRO stage 3, a sharded data parallelism technique that alleviates the memory bottleneck when training large models.  \n",
+    "\n",
+    "In addition, we will utilize a the **SMDDP library**, a handy SageMaker feature which accelerates training by speeding up GPU communication between instances.  The SMDDP Library is compatible with ml.p4d.24xlarge and ml.p4de.24xlarge instances.  For this example, we will use 2 ml.p4d.24xlarge instances, which come with 8x NVIDIA A100 40GB GPUs. \n",
+    "\n",
+    "*Note 1: For the purpose of this example, we will use a dummy synthetic dataset to avoid dealing with an access token required to initialize a Llama2 tokenizer.  This example can be easily modified to supply your own dataset if you own a Llama2 access token*\n",
```

Comment:
nit: We probably need a `.` at the end of the sentence.
- ditto the next line for `note 2`.